map_model/src/objects/turn.rs

```diff
@@ -110,7 +110,14 @@ impl Turn {
     // TODO What should this be for zero-length turns? Probably src's pt1 to dst's pt2 or
     // something.
     pub fn angle(&self) -> Angle {
-        self.geom.first_pt().angle_to(self.geom.last_pt())
+        if self.geom.points().len() < 3 {
+            return Angle::ZERO;
+        }
+        //this is ridiculous, using segments of the turn geometry rather than the start and end angles
```

Comment:
Are you saying we'd get better results or it'd be more clear to use the origin and destination lane's geometry instead?

Reply:
Since the turn geometry is converted from a bezier curve into 5 line segments the angle of the first and last line segment don't completely match the angle of the roads - that's why the turn arrows never seem to be quite turned enough. One possibility would be to store the angle in Turn, but that feels janky.

I'd like to think about not storing turn geometry at all and replacing the intersection in conflicts_with to something like a table lookup, then using curvy turns only for animation. That seems like a big change beyond the scope of this, though.

Reply:
Thanks for the explanation.

> I'd like to think about not storing turn geometry at all and replacing the intersection in conflicts_with to something like a table lookup

IIRC, prior profiling showed that `conflicts_with` can be a hot-spot. A table lookup could have performance benefits as well. I agree we shouldn't tackle that in this PR though.

I'd vote for the conflict table to be built lazily, probably per intersection. If we tried to do it all upfront after loading a map, it'll be slow on the web, since we don't have multithreading working there yet. #786 is related -- we can calculate everything about turns as needed from other map structures, but it makes web slower